src/stardict_lib.cpp

```diff
@@ -856,13 +872,38 @@ bool SynFile::load(const std::string &url, gulong wc)
 
 bool SynFile::lookup(const char *str, glong &idx)
 {
-    glib::CharStr lower_string{ g_utf8_casefold(str, -1) };
-    auto it = synonyms.find(get_impl(lower_string));
-    if (it != synonyms.end()) {
-        idx = it->second;
-        return true;
+    bool bFound = false;
+    glong iTo = synlist.size() - 2;
+    if (iTo <0) return false;
+
+    if (stardict_strcmp(str, get_key(0)) < 0) {
```

Comment:
This is not equivalent of g_utf8_casefold that used before.
stardict_strcmp is just "case independent" for Engish only.
While g_utf8_casefold is "case independent" for all languages that were represented in Unicode

Reply:
This is the same comparison used by [WordListIndex](https://github.com/Dushistov/sdcv/blob/b66799f3584473976915d4321a90befd9309bc52/src/stardict_lib.cpp#L803), and the synonym file should follow the same sorting/searching. According to the stardict [format doc](https://github.com/huzheng001/stardict-3/blob/e51e6656e6b693554853d37a4b36b92fb39b9f82/dict/doc/StarDictFileFormat#L234) the syn file should be sorted with ```stardict_strcmp``` so a binary search should use the same comparison.